src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -147,15 +148,20 @@ public void Update()
                 bool inBeaconControl = _beaconSyncStrategy.ShouldBeInBeaconModeControl();
                 (UInt256? peerDifficulty, long? peerBlock) = ReloadDataFromPeers();
                 // if there are no peers that we could use then we cannot sync
-                if (peerDifficulty is null || peerBlock is null || peerBlock == 0)
+                if (peerBlock is null or 0)
```

Comment:
Changes in this file basically allow `Snapshot` to have peers with nullable `TD`.